routes/preview.js

```diff
@@ -1,228 +1,236 @@
-const path = require('path');
-const stream = require('stream');
-const fs = require('fs');
-const url = require('url');
-const querystring = require('querystring');
 const http = require('http');
-const https = require('https');
-const crypto = require('crypto');
-const zlib = require('zlib');
-const child_process = require('child_process');
 const mime = require('mime');
 
-const {getObject, putObject} = require('../aws.js');
-const puppeteer = require('puppeteer');
+const {getObjectOrNull, putObject} = require('../aws.js');
 const browserManager = require('../browser-manager.js');
+const {appURL} = require('../const.js');
+const {
+  makePromise,
+  timeoutCallback,
+} = require('../lib/async.js');
+const {
+  handle404,
+  handleOptions,
+  promisifyRequest,
+  setHeaders,
+} = require('../lib/request.js');
+const { getURLParams } = require( '../lib/utils.js' )
 
 const PREVIEW_HOST = '127.0.0.1';
 const PREVIEW_PORT = 8999;
 
-const bucketNames = {
-  preview: 'preview.exokit.org',
-};
+const bucketNames = { preview: 'preview.exokit.org' };
 const storageHost = 'https://ipfs.exokit.org';
 
-const _makePromise = () => {
-  let accept, reject;
-  const p = new Promise((a, r) => {
-    accept = a;
-    reject = r;
-  });
-  p.accept = accept;
-  p.reject = reject;
-  return p;
-};
-
-const _warn = err => {
-  console.warn('uncaught: ' + err.stack);
-};
-process.on('uncaughtException', _warn);
-process.on('unhandledRejection', _warn);
+const idRegex = /^\/([0-9]+)/;
+const queryA = /^\/\[([^\]]+)\.([^\].]+)]\/([^.]+)\.(.+)$/;

```

Comment:
Could use a better name.

Reply:
This query is for the `[https://]` URL case.

Reply:
In general these regexes should not be constantized like this.

They are highly tied to the URL lookup functions they perform and are not used anywhere else.